test/util/util.go

```diff
@@ -167,7 +176,7 @@ func ExpectWorkloadsToBePending(ctx context.Context, k8sClient client.Client, wl
 				continue
```

Comment:
For follow up: we shouldn't need the `FindConditionIndex` function. We can use the meta helpers.

Reply:
I have already worked on that at #705.

Reply:
ah, true. Let me check that one now.